sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala

```diff
@@ -114,6 +116,8 @@ object QueryTest {
         Row.fromSeq(s.toSeq.map {
           case d: java.math.BigDecimal => BigDecimal(d)
           case b: Array[Byte] => b.toSeq
+          case d: Double if !d.isNaN && !d.isInfinity => 
+            BigDecimal(d).setScale(10, BigDecimal.RoundingMode.HALF_UP)
```

Comment:
Instead of changing how we compare double values, how about we change our tests by casting results to the decimal type?